### PR TITLE
[WiP] Feature : better cover filename handling

### PR DESCRIPTION
**Pull request by Go-rom**
* [x] I am the author of this code or the code is public domain
* [x] I release this code into the public domain

Hello,
first you have to know that this PR is my first "real" fix. I'm not a dev, so far my PR were mainly text fixes and translations on several projects.
I have recently read https://github.com/PodcastGenerator/PodcastGenerator/issues/338 and this issue is also bothering me (can't change cover with Spotify Podcasts), so this PR aims to fix it.

**PR details**

When a new cover is uploaded, the original filename is kept as the new podcast name instead of the static name "itunes_image.jpg"
Also at first setup of Podcast Generator, config.php now generates a new variable : 'podcast_cover' with "itunes_image.jpg' as first value. When a new cover will be uploaded, the filename will be stored in this variable.

In case of a new episode which doesn't embed a cover art, it will displays the global podcast cover.

**This PR doesn't provide**
- Filename editing with a basic textfield, it only uses file selector.
- Multiple covers for episodes and seasons. It's a good feature request but needs a specific PR

**It may lack**
- Some failsafe code in order to prevent unwanted filenames
- Some code to support PG upgrades

The branch has been tested with a fresh install and it works.
Also, i tried to know if not using "itunes_image.jpg" could be a problem but i didn't find anything about it.

Again, i'm not a dev so please take a deep look at this changes and provide me feedbacks, it's a good way for me to experience "easy hacks"

Task list :
- [ ] Fixes/improves based on feedbacks
- [ ] re-run a test with a fresh install
- [ ] wait for test results from other users
- [ ] mark this PR as ready to be merged

### CODE REVIEW

**Comment by Go-rom** [comment]
I would like to go further about the upload feature.

Here are some thoughts : 

- Add PNG support (Global podcast requirements say that JPG and PNG are commonly accepted)
- Set min resolution and max resolution in pixels (Global podcast requirements start from 1400px to 3000 px)
- Check pixel density (Global podcast requirement : 72 dpi)
- Check if the file 
- Check filesize, must no be 0. I don't think that setting a max size is relevant
- Limit range of characters (a-z, 0-9; "-", "_") and enforce lowercase letters
- Check number of characters (i would say max. 20)
- Check if filename already exists (if user keeps the same filename, new cover art won't be detected by some podcast platform like Spotify)

Here are iTunes podcast requirements : https://help.apple.com/itc/podcasts_connect/#/itcb54353390

Tell me what you think about these suggestions, thank you

**Comment by emilengler** [comment]
Thanks for working on this! We have already support for episode specific covers! I am just not sure if it was already included in 3.1 though

**Comment by Go-rom** [comment]
> We have already support for episode specific covers! 

Yes but this PR is only about Podcast cover art. Currently, the artwork is saved as "itunes_image.jpg" and this choice is hard-coded. I checked your current master branch and didn't see anything about that.

This is related to https://github.com/PodcastGenerator/PodcastGenerator/issues/338
If you're already tackled this issue, let me know and i will close this PR. Thanks !

File: PodcastGenerator/themes/default/singleepisode.php [hunk 1]
```diff
@@ -48,10 +48,10 @@
     echo '  <img class="card-img" src="' . $filename . '">';
     echo '  </div>';
 }
-// If no cover art, we use itunes_image.jpg
+// If no episode cover art, we use the global feed covert art
```
Comment (coldacid) [inline PodcastGenerator/themes/default/singleepisode.php:51]: Typo: "covert" -> "cover"

File: PodcastGenerator/core/backwards.php [hunk 2]
```diff
@@ -69,6 +69,8 @@ function backwards_3_1_to_3_2($absoluteurl)
 
 \$podcast_description = \"".$config['podcast_description']."\";
 
+\$podcast_cover = \"".$config['podcast_cover']."\";
```
Comment (coldacid) [inline PodcastGenerator/core/backwards.php:72]: Check to see if `$config['podcast_cover']` has a value, and write out itunes_image.jpg if it doesn't:

```php
\$podcast_cover = \"".(isset($config['podcast_cover']) ? $config['podcast_cover'] : 'itunes_image.jpg')."\";
```

**Review by coldacid — CHANGES_REQUESTED** [review]

**Pull request merged**